File: izayoi-site/src/main/webapp/scripts/izayoi.js
/*
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF
 *
 * ANY KIND, either express or implied. See the License for the specific language governing rights and
 *
 * limitations under the License.
 *
 * The Original Code is the IZAYOI web framework.
 *
 * The Initial Developer of the Original Code is
 *
 *   Mo Chen <withinsea@gmail.com>
 *
 * Portions created by the Initial Developer are Copyright (C) 2009-2010
 * the Initial Developer. All Rights Reserved.
 */

$(function () {


    // external links

    $('a[href^=http://], a[href^=https://], a[href^=ftp://]').addClass('external').attr('target', "_blank");
    $('a[href^=mailto:]').addClass('mailto');


    // code beautify

    var FORMAT_TYPES = {
        'js': 'javascript',
        'json': 'javascript',
        'xml': 'xml',
        'html': 'html',
        'jsp' : 'java'
    };

    $('pre.code > code[src]').each(function () {
        var code = $(this), pre = code.parent();
        var split = code.attr('src').split(':');
        var type = FORMAT_TYPES[split[1].toString().replace(/^.*\./, '')] || 'plain';
        $.get(split[0], {}, function (props) {
            var content = props[split[1]].replace(/</g, '&lt;').replace(/</g, '&gt;');
            $('<pre class="code"><code>' + content + '</code></pre>').insertAfter(pre).beautifyCode(type);
            pre.remove();
        }, 'properties');
    });
});

File: izayoi-bundle/src/main/java/org/withinsea/izayoi/Izayoi.java
/*
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF
 *
 * ANY KIND, either express or implied. See the License for the specific language governing rights and
 *
 * limitations under the License.
 *
 * The Original Code is the IZAYOI web framework.
 *
 * The Initial Developer of the Original Code is
 *
 *   Mo Chen <withinsea@gmail.com>
 *
 * Portions created by the Initial Developer are Copyright (C) 2009-2010
 * the Initial Developer. All Rights Reserved.
 */

package org.withinsea.izayoi;

import org.withinsea.izayoi.cloister.Cloister;
import org.withinsea.izayoi.commons.servlet.ServletFilterUtils;
import org.withinsea.izayoi.core.conf.IzayoiContainer;
import org.withinsea.izayoi.core.conf.IzayoiContainerFactory;
import org.withinsea.izayoi.cortile.Cortile;
import org.withinsea.izayoi.glowworm.Glowworm;

import javax.servlet.*;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.Map;

/**
 * Created by Mo Chen <withinsea@gmail.com>
 * Date: 2010-3-6
 * Time: 10:41:06
 */
public class Izayoi implements Filter {

    protected Cloister cloister;
    protected Glowworm glowworm;
    protected Cortile cortile;

    @Override
    public void init(FilterConfig filterConfig) throws ServletException {

        ServletContext servletContext = filterConfig.getServletContext();
        Map<String, String> overriddenProperties = ServletFilterUtils.getParamsMap(filterConfig);

        IzayoiContainer container = new IzayoiContainerFactory()
                .addModule("org.withinsea.izayoi.core")
                .addModule("org.withinsea.izayoi.cloister")
                .addModule("org.withinsea.izayoi.glowworm")
                .addModule("org.withinsea.izayoi.cortile")
                .create(servletContext, overriddenProperties);

        cloister = new Cloister();
        glowworm = new Glowworm();
        cortile = new Cortile();

        cloister.init(container);
        glowworm.init(container);
        cortile.init(container);
    }

    @Override
    public void doFilter(ServletRequest req, ServletResponse resp, final FilterChain chain) throws IOException, ServletException {
        ServletFilterUtils.chain((HttpServletRequest) req, (HttpServletResponse) resp, chain, cloister, glowworm, cortile);
    }

    @Override
    public void destroy() {
    }
}